Rename ofxgo_test package to ofxgo and remove self-imports/references

JohnStarich opened the pull request:
It turned out that [removing the self-imports](https://github.com/aclindsa/ofxgo/pull/23#issuecomment-606048408) was more involved than I previously thought! This PR renames the test package to use the same name (`ofxgo`) so imports aren't required to run the tests. I've also un-exported `ErrInvalid` so it is no longer part of the API surface area.

Luckily `sed` is amazing, so 99% of the renaming and import deletion effort was automatic 😅 

Since this touches so many lines in the library, I figured it'd be worth breaking this out into it's own PR, for our sanity.

aclindsa commented on constants_test.go:
```diff
@@ -1,4 +1,4 @@
-package ofxgo_test
```
I believe generate_constants.py should also be updated, since I'm lazy and use it to generate constants.go and constants_test.go if the constants change.

JohnStarich commented:
Updated! I had trouble generating the constants locally, probably due to differences in my version of python 🤷‍♂️

aclindsa commented on generate_constants.py:
```diff
@@ -164,7 +164,7 @@
+test_header = """package ofxgo
 
 /*
  * Do not edit this file by hand. It is auto-generated by calling `go generate`.
```
Sorry I wasn't more explicit earlier. I believe `test_header` also needs to have the `"github.com/aclindsa/ofxgo"` import line removed, as well as several `ofxgo.` prefixes in `test_template`.

aclindsa commented:
> Updated! I had trouble generating the constants locally, probably due to differences in my version of python man_shrugging

What errors did you see (wondering if this is something that should be fixed)?

JohnStarich commented:
Nevermind! Turned out it was a problem with my `#!/usr/bin/env python` running Python 2 instead of 3. (Also managed to break, then unbreak my python install in the process... 🎉)

This is why things seemed busted to me: https://stackoverflow.com/a/27262274/1530494
All of the dict items for generating the file were in a very different order, but switching to Py3 fixed it 👍

JohnStarich commented:
As an aside, it might be useful to generate these Go files with another Go script, like another `cmd` package utility. That way the tooling and setup is a tad more consistent for new contributors 👌

aclindsa commented:
Looks good to me, but now it's complaining about merge conflicts!

JohnStarich commented:
Oh weird, it looks like it's clean to me! 
<img width="486" alt="image" src="https://user-images.githubusercontent.com/1009441/78208902-29cf5e00-746b-11ea-8d9e-0feccec538c7.png">

aclindsa commented:
> Oh weird, it looks like it's clean to me!

Hah, figured out this is dependent upon the merge strategy. I only see this when I have 'Rebase and merge' selected (which was selected by default, probably because that's what I selected most recently).

JohnStarich commented:
I've just rebased, squashed, and force-pushed (the force-push also links to the diff, copying [here](https://github.com/aclindsa/ofxgo/compare/572006f3440c2c912f5055ea4e87921edb9b663b..ada0c0d0108ade3855eb185183a3509422cb0bcf)) so now it shouldn't complain 😄

JohnStarich commented:
Thanks Aaron! Would you mind cutting a release tag on that one?

Thanks so much for all the timely reviews as always, much appreciated 🚀
